beacon-chain/blockchain/chain_info.go

```diff
@@ -60,25 +59,27 @@ type ParticipationFetcher interface {
 
 // FinalizedCheckpt returns the latest finalized checkpoint from head state.
 func (s *Service) FinalizedCheckpt() *ethpb.Checkpoint {
-	if s.headState == nil || s.headState.FinalizedCheckpoint() == nil {
+	if s.finalizedCheckpt == nil {
 		return &ethpb.Checkpoint{Root: params.BeaconConfig().ZeroHash[:]}
 	}
-	cpt := s.headState.FinalizedCheckpoint()
+
+	cpt := state.CopyCheckpoint(s.finalizedCheckpt)
```

Comment:
Can you move this beneath the equality check and just compare s.finalizedCheckpt.Root in bytes.Equal? Might save some copies

Reply:
Good suggestion. Done